api/web3server.go

```diff
@@ -0,0 +1,870 @@
+		return nil, err
+	}
+	idxStr, err := getStringFromArray(in, 1)
+	if err != nil {
+		return nil, err
+	}
+	idx, err := hexStringToNumber(idxStr)
+	if err != nil {
+		return nil, err
+	}
+	actionInfos, err := svr.coreService.ActionsByBlock(removeHexPrefix(blkHash), idx, 1)
+	if err != nil {
+		return nil, err
+	}
+	return svr.getTransactionCreateFromActionInfo(actionInfos[0])
+}
+
+func (svr *Web3Server) getTransactionByBlockNumberAndIndex(in interface{}) (interface{}, error) {
+	blkNum, err := getStringFromArray(in, 0)
+	if err != nil {
+		return nil, err
+	}
+	idxStr, err := getStringFromArray(in, 1)
+	if err != nil {
+		return nil, err
+	}
+	num, err := svr.parseBlockNumber(blkNum)
+	if err != nil {
+		return nil, err
+	}
+	idx, err := hexStringToNumber(idxStr)
+	if err != nil {
+		return nil, err
+	}
+	log.L().Error("acquire block meta")
+	blkMetas, err := svr.coreService.BlockMetas(num, 1)
+	if err != nil || blkMetas == nil || len(blkMetas) == 0 {
+		if err == nil {
+			err = errors.New("invalid block")
+		}
+		return nil, err
+	}
+	actionInfos, err := svr.coreService.ActionsByBlock(blkMetas[0].Hash, idx, 1)
+	if err != nil || len(actionInfos) == 0 {
+		log.L().Error("index err", zap.Error(err), zap.Int("actLen", len(actionInfos)))
+		return nil, err
+	}
+	log.L().Error("extract action info")
+	return svr.getTransactionCreateFromActionInfo(actionInfos[0])
+}
+
+func (svr *Web3Server) newFilter(filter *filterObject) (interface{}, error) {
+	//check the validity of filter before caching
+	if filter == nil {
+		return nil, errNullPointer
+	}
+	if _, err := svr.parseBlockNumber(filter.FromBlock); err != nil {
+		return nil, errors.Wrapf(errUnkownType, "from: %s", filter.FromBlock)
+	}
+	if _, err := svr.parseBlockNumber(filter.ToBlock); err != nil {
+		return nil, errors.Wrapf(errUnkownType, "to: %s", filter.ToBlock)
+	}
+	for _, ethAddr := range filter.Address {
+		if _, err := ethAddrToIoAddr(ethAddr); err != nil {
+			return nil, err
+		}
+	}
+	for _, tp := range filter.Topics {
+		for _, str := range tp {
+			if _, err := hex.DecodeString(str); err != nil {
+				return nil, err
+			}
+		}
+	}
+
+	// cache filter and return hash value of the filter as filter id
+	filter.FilterType = "log"
+	objInByte, _ := json.Marshal(*filter)
+	keyHash := hash.Hash256b(objInByte)
+	filterID := hex.EncodeToString(keyHash[:])
+	err := svr.cache.Set(filterID, string(objInByte))
+	if err != nil {
+		return nil, err
+	}
+	return "0x" + filterID, nil
+}
+
+func (svr *Web3Server) newBlockFilter() (interface{}, error) {
+	filterObj := filterObject{
+		FilterType: "block",
+		LogHeight:  svr.coreService.bc.TipHeight(),
+	}
+	objInByte, _ := json.Marshal(filterObj)
+	keyHash := hash.Hash256b(objInByte)
+	filterID := hex.EncodeToString(keyHash[:])
+	err := svr.cache.Set(filterID, string(objInByte))
+	if err != nil {
+		return nil, err
+	}
+	return "0x" + filterID, nil
+}
+
+func (svr *Web3Server) uninstallFilter(in interface{}) (interface{}, error) {
```

Comment:
`unregisterFilter` or `removeFilter`?

Reply:
the method's name is `eth_uninstallFilter`, uninstallFilter is straightforward